recipes-bsp/u-boot/u-boot-iot2050.inc

```diff
@@ -46,28 +46,38 @@ DEBIAN_BUILD_DEPENDS_append_secureboot = ", python3-pycryptodome:native, \
 DEPENDS_append_otpcmd = " secure-boot-otp-provisioning"
 DEBIAN_BUILD_DEPENDS_append_otpcmd = ", secure-boot-otp-provisioning"
 
+TEMPLATE_FILES += "rules.tmpl"
+TEMPLATE_VARS += "FIRMWARE_SECURE_VER"
+
 U_BOOT_CONFIG_PACKAGE = "1"
 
 do_prepare_build_append() {
-    cd ${S}
-    ln -sf ../prebuild/* ${S}
+    ln -f ${WORKDIR}/prebuild/* ${S}
     ln -sf /lib/firmware/k3-rti-wdt.fw ${S}
     echo "${SPI_FLASH_IMG} /usr/lib/u-boot/${MACHINE}" > \
         ${S}/debian/u-boot-${MACHINE}.install
     echo -$(${ISAR_RELEASE_CMD}) > ${S}/.scmversion
     cp ${WORKDIR}/rules ${S}/debian/rules
-    rm -f ${S}/keys
+    rm -rf ${S}/keys
 }
 
 do_prepare_build_append_secureboot() {
     if [ "${PRODUCT_GENERATION}" = "pg1" ]; then
         bbwarn "PG1 devices are not supported in booting securely"
     fi
-    ln -s ../keys ${S}
+    mkdir -p ${S}/keys
+    ln -f ${WORKDIR}/keys/* ${S}/keys/
     sed -ni '/### Secure boot config/q;p' ${S}/configs/${U_BOOT_CONFIG}
     cat ${WORKDIR}/secure-boot.cfg >> ${S}/configs/${U_BOOT_CONFIG}
 }
 
+dpkg_runbuild_append_secureboot() {
+    # remove private keys from source archive
```

Comment:
We need a better pattern for that. Maybe the keys can be injected into the overlay of the schroot by using a ` -chroot-setup-commands=` (currenty ISAR has no support for that).

For now, I can live with it. But should be discussed on the ISAR ML.